src/sql/workbench/browser/modal/modal.ts

```diff
@@ -622,3 +715,11 @@ export abstract class Modal extends Disposable implements IThemable {
 		this._footerButtons = [];
 	}
 }
+
+function getThemeColor(themeColorName: string): Color {
```

Comment:
This function doesn't seem to make sense - you're registering a callback here but not calling it so you're always going to be returning undefined

Reply:
Oh it works alright. It receives a theme color name and returns a color value. Would you like me to add code comments to that effect?

Reply:
I don't know how it could possibly work. registerThemingParticipant is taking a callback function that will be invoked whenever the theme is changed. The only way that themeColor wouldn't be undefined here is if the callback was invoked right away - but even if that somehow happened that'd only be because the timing worked out by luck.

I assume what you want to do here is get the current color for a theme - you should be using `themeService.getColorTheme().getColor(themeColorName)` in that case.

Reply:
Thanks for that! I have updated the code accordingly.